### PR TITLE
Dasher types

### PR DESCRIPTION
**Pull request by KeanW**
Please fill in this template.

- [ ] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/forge-viewer/index.d.ts [hunk 1]
```diff
@@ -26,2432 +27,2513 @@
...
+      const CONTROL_ADDED = 'ControlGroup.ControlAdded';
+      const CONTROL_REMOVED = 'ControlGroup.ControlRemoved';
+      const SIZE_CHANGED = 'ControlGroup.SizeChanged';
+      const STATE_CHANGED = 'Button.StateChanged';
+
+      class DockingPanel {
+        constructor(parentContainer: HTMLElement, id: string, title: string, options?: DockingPanelOptions);
+
+        closer: HTMLElement;
+        container: any;
```
Comment (liskaj) [inline types/forge-viewer/index.d.ts:1788]: Should this be really any?

File: types/forge-viewer/index.d.ts [hunk 2]
```diff
@@ -26,2432 +27,2513 @@
...
+        content: any;
```
Comment (liskaj) [inline types/forge-viewer/index.d.ts:1789]: Same as above.

File: types/forge-viewer/index.d.ts [hunk 3]
```diff
@@ -5,6 +5,7 @@
 //                 Jan Liska <https://github.com/liskaj>
 //                 Petr Broz <https://github.com/petrbroz>
 //                 Cyrille Fauvel <https://github.com/cyrillef>
+//                 Kean Walmsley <https://github.com/keanw>
```
Comment (liskaj) [inline types/forge-viewer/index.d.ts:8]: Do you want to be owner? This means approving PRs from others and sometime reacting on feedback from DT community. Note you can still submit PR w/o being owner. Also note that according to DT rules owner can't approve own changes.

File: types/forge-viewer/index.d.ts (hunk 1 above)
Comment (KeanW) [reply types/forge-viewer/index.d.ts:1788]: The main issue is that dockRight and dockBottom properties aren't part of HTMLElement, and these are very much needed by Forge viewer apps using panels (well, Dasher, anyway). I'd be happy if there were an easy way to augment the HTMLElement type with these properties. I just defaulted to any, which I agree isn't ideal.

File: types/forge-viewer/index.d.ts (hunk 2 above)
Comment (KeanW) [reply types/forge-viewer/index.d.ts:1789]: See above.

File: types/forge-viewer/index.d.ts (hunk 3 above)
Comment (KeanW) [reply types/forge-viewer/index.d.ts:8]: No no - I'm not sure why that was included in the request. Fat fingers, probably.

Comment (liskaj) [reply types/forge-viewer/index.d.ts:8]: Ok, then just remove it and I'll approve.

Comment (KeanW) [reply types/forge-viewer/index.d.ts]: Done!

**Review by liskaj — APPROVED** [review]
👍

**Pull request closed without merging**